### PR TITLE
test: e2e tests with java 21

### PR DESCRIPTION
**Pull request by jeromevdl**
**Issue #, if available:**

## Description of changes:

Add support for Java 21 in end-to-end tests.

**Checklist**

 

* [x] [Meet tenets criteria](https://docs.powertools.aws.dev/lambda-java/#tenets)
* [x] Update tests
* [ ] Update docs
* [x] PR title follows [conventional commit semantics](https://www.conventionalcommits.org/en/v1.0.0/)

## Breaking change checklist

**RFC issue #**:

* [ ] Migration process documented
* [ ] Implement warnings (if it can live side by side)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by scottgerring — APPROVED** [review]
The parts of this you wrote @jeromevdl look right to me, and i've obviously spent some time testing them now. Not sure if we should hold off merging until we get the AspectJ upstream merge which seems eminent, as we'd just roll most of this back then.

**Review by scottgerring — APPROVED** [review]

File: README.md
Comment (mriccia) [inline README.md:17]: What exactly does this mean? Do we have examples?

Comment (scottgerring) [reply README.md]: cleaned up wording a bit. Hopefully AspectJ for 21 will be released soon and we can remove this again!

**Pull request merged**
